M2/Macaulay2/packages/SuperLinearAlgebra.m2

```diff
@@ -0,0 +1,689 @@
+       Email => "tarashikashanifatemeh@gmail.com", 
+       HomePage => "Your page here"}
+   },
+   Headline => "Berezinian and supertrace of a supermatrix",
+   DebuggingMode => true
+)
+
+--------------------
+--Exports
+--------------------
+
+export {
+    --Types
+        "SuperMatrix",
+    --Methods
+        "Berezinian",
+        "inverseSuperMatrix",
+        "Parity",
+        "superMatrix",
+        "superMatrixParity",
+        "superRing",
+        "superTrace",
+    --Symbols
+        "supermatrix",
+        "targetM1",
+        "targetM3", 
+	"sourceM1", 
+	"sourceM2"
+ }
+
+--------------------------------------------------------------
+--SuperRing (Super commutative ring) 
+--------------------------------------------------------------
+superRing = method();
+superRing (PolynomialRing,PolynomialRing):= (R1,R2) -> (
+         n := #gens R1;
+         inverseVariable := symbol inverseVariable;
+ 	 R11 := symbol R11; 
+	 R22 := symbol R22; 
+	 R111 := symbol R111;
+         R11 = (coefficientRing R1)[R1_0..R1_(n-1),inverseVariable_0..inverseVariable_(n-1)];
+	 R111 = R11/apply(0..(n-1), i -> sub(R1_i,R11) * inverseVariable_i - 1);
+         m := #gens R2;
+         w := (for i to m-1 list (0))|toList(0..(m-1));
+         R22 = (coefficientRing R2)[R2_0..R2_(m-1), MonomialOrder=>{Weights => w,Lex}, SkewCommutative=>true];
+         print concatenate {"is a super commutative ring of dimension", toString n, "|",toString m};
+         R111**R22
+         )         
+ 
+------------------------------------------------------
+--SuperMatrix
+--This a is new mulitivariate Hash table with 5 keys.
+--supermatrix, targetM1, targetM3, sourceM1, sourceM2
+------------------------------------------------
+SuperMatrix = new Type of MutableHashTable;
+
+superMatrix = method();
+superMatrix (Matrix,Matrix,Matrix,Matrix):= (M1,M2,M3,M4) ->(
+         nr1 := numgens target M1;
+         nr2 := numgens target M3;
+         ns1 := numgens source M1;
+         ns2 := numgens source M2;
+	 T1 := symbol T1;
+	 T2 := symbol T2;
+         T1 = M1 | M2;
+         T2 = M3 | M4;
+	 new SuperMatrix from{
+	 supermatrix => T1 || T2,
+	 targetM1 => nr1,
+	 targetM3 => nr2,
+	 sourceM1 => ns1,
+	 sourceM2 => ns2
+	 }
+         )
+
+TEST ///
+M1 = matrix {{1,2},{5,6},{9,10}};
+M2 = matrix {{3,4},{7,8},{11,12}};
+M3 = matrix {{13,14},{17,18}};
+M4 = matrix {{15,16},{19,20}};
+G = superMatrix(M1,M2,M3,M4);
+G.supermatrix;
+assert(G.supermatrix == matrix {{1,2,3,4},{5,6,7,8},{9,10,11,12},{13,14,15,16},{17,18,19,20}})
+assert(G.targetM1 == 3)
+assert(G.targetM3 == 2)
+assert(G.sourceM1 == 2)
+assert(G.sourceM2 == 2)
+///
+
+--------------------
+--Parity  
+-------------------- 
+Parity = method();
+Parity (RingElement,Ring,List) := (f,R,a) -> (
+    e := symbol e;
+    e = exponents f;
+    l := symbol l;
+    l={};
+    for i from 0 to (#gens R -1) do (for j from 0 to #a -1 do (if R_(i)==a_(j) then (l= append(l,i))));
+    d := symbol d;
```

Comment:
This assignment is superfluous, because the value is never used.  Check for other instances of this kind of thing.

Reply:
You could delete this line and change `d=0` below to `d:=0`.